### PR TITLE
Add open orders

### PR DESCRIPTION
**Pull request by Tsopic**

### CODE REVIEW

**Comment by Tsopic** [comment]
So needs more testing, @iamfotx will take it over from here

File: src/tests/unit/api-next/get-all-open-orders.test.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          "equityWithLoanBefore": 1.7976931348623157e+308,
+          "initMarginChange": 1.7976931348623157e+308,
+          "maintMarginChange": 1.7976931348623157e+308,
+          "equityWithLoanChange": 1.7976931348623157e+308,
+          "initMarginAfter": 1.7976931348623157e+308,
+          "maintMarginAfter": 1.7976931348623157e+308,
+          "equityWithLoanAfter": 1.7976931348623157e+308,
+          "commission": 1.7976931348623157e+308,
+          "minCommission": 1.7976931348623157e+308,
+          "maxCommission": 1.7976931348623157e+308,
```
Comment (Tsopic) [inline src/tests/unit/api-next/get-all-open-orders.test.ts:143]: So these should be numbers or undefined now? @mfrener 

cc @iamfotx

**Comment by mfrener** [comment]
@iamfotx
I'm really busy with other work right, so that's why there was no progress on this or the other PRs past weeks, but I'm online 24/7 (almost :D), so just ping me if there any questions (will be back on trading project soon, hopefully ;) )

File: src/tests/unit/api-next/get-all-open-orders.test.ts (hunk 1 above)
Comment (mfrener) [reply src/tests/unit/api-next/get-all-open-orders.test.ts:143]: Yes, on IBApiNext there shall be no MAX_NUMBER but should be undefined (in contrast to IBApi).
Think we should add a helper function for this, as we will need it on other place too.
Something like:
```
undefineMaxAttributes(obj: unknown): void {
  Object.keys(obj).forEach(key => {
     if (typeof (<Record<String,Number>>obj)[key]=== 'number') {
         (<Record<String,Number>>obj)[key] = undefineMax(obj);
     }
  });
}
```
and use it to undefine MAX_NUMBER 's on the given object(s)

**Comment by Tsopic** [comment]
https://github.com/stoqey/ib/pull/118 resolves it, closing

**Pull request closed without merging**